dap-mode.el

```diff
@@ -1834,10 +1837,12 @@ point is set."
   :init-value nil
   :global t
   :group 'dap-mode
+  (declare-function dap-ui-mode "dap-ui" (&optional arg))
```

Comment:
Same here, `declare-function` should be moved to top level (not inside any function/macro).
And actually `dap-ui-mode` is autoloaded so it will not require `declare-function`.
Only the `dap-ui-many-windows-mode` is require to be `declare-function`.

The rule of thumb is that unless byte-compile complains, we normally don't need `declare-function`.

Reply:
That is the point - there were byte-compile warnings. declare-funtion, like the defvar above, has no overhead (zero: not a byte of bytecode). It is a compile-time hint so that the compiler would not throw warnings.

Reply:
Is there byte-compile warning for `dap-ui-mode`? Which is autoloaded?

Reply:
Yes. The byte compiler cannot know that some function, which is not loaded at global scope (that is, not even locally in a defun) be defined, and so will throw warnings. For such things, declare-function is made for: it tells the byte compiler that one is sure that some function will be available at run time, and optionally, what its signature is, eliminating such warnings.